packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -216,44 +264,53 @@ async function visualizeObjectNode(
 async function visualizeMapNode(
 	tree: VerboseTreeNode,
 	nodeSchema: SimpleMapNodeSchema,
-	treeSchema: SimpleTreeSchema,
+	treeDefinitions: ReadonlyMap<string, SimpleNodeSchema>,
+	isRequired: boolean | undefined,
 	visualizeChildData: VisualizeChildData,
 ): Promise<VisualSharedTreeNode> {
+	const mapNodeSchemaProperties: Record<string, FieldSchemaProperties> = {};
+
+	for (const key of Object.keys(tree.fields)) {
+		mapNodeSchemaProperties[key] = {
+			allowedTypes: nodeSchema.allowedTypes,
+			isRequired: undefined,
+		};
+	}
 	return {
 		schema: {
 			schemaName: tree.type,
-			allowedTypes: `Record<string, ${concatenateTypes(nodeSchema.allowedTypes)}>`,
+			allowedTypes: concatenateTypes(nodeSchema.allowedTypes),
```

Comment:
I think we have an issue here. For object nodes (in `visualizeObjectNode`), the allowed types are the ones that are passed in, i.e. the ones allowed in the field that _contains_ the object node. Or in other words, what the object node _itself_ could have been. But for map nodes, in that same place we are displaying the allowed types for the _elements_ of the map, instead of what the map node itself could have been? E.g. in a schema like this:

```typescript
new TreeViewConfiguration({ schema: [SchemaFactory.number, SchemaFactory.map(schemaFactory.string)] });
```

Where the root can be a number or a map of string, if we put a map node at the root, I think we end up with it showing `allowedTypes => string`, instead of `number | Map`?

Reply:
That's a great catchk, thanks! I made the fix and refactor on both map and array node visualizers. The visualizer now renders `allowedTypes` for the schema below:

```ts
		class TodoWorkspace extends builder.object("todo-workspace", {
			categories: builder.object("todo-categories", {
				work: [builder.map([WorkItem]), builder.array(WorkItem)],
				personal: [builder.map([PersonalItem]), builder.array(PersonalItem)],
			}),
		}) {}
```

![Screenshot 2025-02-12 at 17 04 08](https://github.com/user-attachments/assets/a030046e-a5a3-41a5-abc4-4ef8c51473b5)


Will verify the test results again to make sure the logic is correct.